### PR TITLE
Use ac-fus-lib as npm package

### PR DESCRIPTION
**Pull request by annakocheshkova**
It was decided to treat mc-fus as a separate library and publish it to npm without creating a separate repo. 
This PR makes it a proper library and moves to the root of the repo (for visibility).
Also did some minor renaming to differentiate it with the original mc-fus library for web.

This PR also adds an additional step to CI that builds the library.
[AB#81826](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/81826)

### CODE REVIEW

**Comment by jp-andre** [comment]
> It was decided to treat mc-fus as a separate library and publish it to npm without creating a separate repo.

I don't feel very comfortable with this approach. Why not create a separate open source repo?
If this lib can be used in another client, it also shouldn't be in the CLI. Could it be shared by the portal? Otherwise, why have a separate npm lib?

File: appcenter-file-upload-client-node/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  inputs:
+    versionSpec: '10.x'
+  displayName: 'Install Node.js'
+
+- script: |
+    npm install
+    npm install -g eslint
+    npm run build
+    npm run test
+  displayName: 'Build and test'
```
Comment (jp-andre) [inline appcenter-file-upload-client-node/azure-pipelines.yml:23]: What is this file for? Are we creating a new pipeline in azure? I don't think this approach can work.

File: package.json [hunk 2]
```diff
@@ -44,7 +45,8 @@
   "files": [
     "package-lock.json",
     "bin",
-    "dist"
+    "dist",
+    "appcenter-file-upload-client-node/out"
```
Comment (jp-andre) [inline package.json:49]: I think this is missing various files, like package.json and package-lock.json.

File: package.json [hunk 3]
```diff
@@ -17,7 +17,8 @@
   },
   "preferGlobal": true,
   "scripts": {
-    "build": "npm run lint && gulp build:raw",
+    "build-ac-fus": "cd appcenter-file-upload-client-node && npm i && npm run build",
```
Comment (jp-andre) [inline package.json:20]: `npm i` should happen during install, not during build

File: appcenter-file-upload-client-node/.eslintignore [hunk 4]
```diff
@@ -0,0 +1,8 @@
+node_modules
+
+# compiled files
+out
+build
+
+# db migration files
+db
```
Comment (jp-andre) [inline appcenter-file-upload-client-node/.eslintignore:8]: newline please

File: appcenter-file-upload-client-node/.prettierignore [hunk 5]
```diff
@@ -0,0 +1,10 @@
+node_modules
+
+# compiled files
+out
+build
+
+# db migration files
+db
+
+random-folder
```
Comment (jp-andre) [inline appcenter-file-upload-client-node/.prettierignore:10]: newline

File: package-lock.json [hunk 6]
```diff
@@ -758,6 +758,1969 @@
...
       }
     },
+    "appcenter-file-upload-client-node": {
+      "version": "file:appcenter-file-upload-client-node",
+      "requires": {
+        "abort-controller": "^3.0.0",
+        "node-fetch": "^2.6.0"
+      },
+      "dependencies": {
+        "@appcenter/eslint-config-appcenter": {
```
Comment (jp-andre) [inline package-lock.json:768]: this is not acceptable: we can't use a private package from an open source repo

**Comment by annakocheshkova** [comment]
@jp-andre Resolved the feedback, following the test instructions now works for me even if I install the `.tgz` package globally.

**Comment by jp-andre** [comment]
That's awesome! Thanks so much for taking the time. We will have another look on Monday (tmr is our day of learning).

**Comment by anywherepilot** [comment]
Works indeed.

**Pull request merged**